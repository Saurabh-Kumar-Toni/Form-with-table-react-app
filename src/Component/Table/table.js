import React, { useState } from 'react';

const Table = ({items, deleteUsers, findUsers}) => {

    //const items = props;
    const [search, setSerach] = useState("");
    const handleDelete = (id) => {

      deleteUsers(id);
    }

    const handleChange = (e) => {
      setSerach(e.target.value);
      findUsers(e.target.value);
    }

    return (
      <div id="Table">

        <input id="searchbar" type = "text" value={search}  name="search"  onChange={(e) => handleChange(e)}
        placeholder="Search in Table"/>
        <table>
          <tbody>
            <tr>
              <th>NAME</th>
              <th>PHONE</th>
              <th>GENDER</th>
              {/* <th>DELETE</th> */}
            </tr>
            {items.map((item, id=0) => {
              return (
                <tr key={id}>
                  <td>{item.username}</td>
                  <td>{item.number}</td>
                  <td>{item.gender}</td>
                  <td><button id="btndel" key={id} name="delete" value="delete" onClick={() => handleDelete(id)}>X</button></td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
    );
  
}

export default Table;