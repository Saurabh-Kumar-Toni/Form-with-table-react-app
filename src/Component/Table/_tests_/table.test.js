import React from 'react';
import { render, screen } from '@testing-library/react';
import Table from '../table';


test('renders the test table component properly', () => {
   

    const mockedFn = jest.fn();
    const item = [{
        username: "Rohan",
        number: "1234567890", 
        gender: "M"
      }];
    const {container} = render(<Table items={item} deleteUsers={jest.fn()} findUsers={jest.fn()}/>)
    expect(container.firstChild).toMatchSnapshot();
  });
  