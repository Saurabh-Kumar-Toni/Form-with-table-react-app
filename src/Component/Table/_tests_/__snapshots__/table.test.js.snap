// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the test table component properly 1`] = `
<div
  id="Table"
>
  <input
    name="search"
    type="text"
    value=""
  />
  <table>
    <tbody>
      <tr>
        <th>
          NAME
        </th>
        <th>
          PHONE
        </th>
        <th>
          GENDER
        </th>
      </tr>
      <tr>
        <td>
          Rohan
        </td>
        <td>
          1234567890
        </td>
        <td>
          M
        </td>
        <td>
          <button
            id="btndel"
            name="delete"
            value="delete"
          >
            X
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
